cmd/maya-apiserver/cstor-operator/storagepool_create.go

```diff
@@ -18,42 +18,25 @@ package spc
 
 import (
 	"github.com/golang/glog"
-	nodeselect "github.com/openebs/maya/pkg/algorithm/nodeselect/v1alpha1"
-	"github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
-	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	apisv1alpha1 "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
 	"github.com/openebs/maya/pkg/storagepool"
 	"github.com/pkg/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// PoolCreateConfig is config object used to create a cstor pool.
-type PoolCreateConfig struct {
-	*nodeselect.Config
-	*Controller
-}
-
-// CasPoolBuilder is the builder object for cas pool.
-type CasPoolBuilder struct {
-	CasPool *apis.CasPool
-}
-
 // Cas template is a custom resource which has a list of runTasks.
 
-// runTasks are configmaps which has defined yaml templates for resources that needs
-// to be created or deleted for a storagepool creation or deletion respectively.
+// runTasks are custom resource which has defined yaml templates for resources that needs
+// to be created for a storagepool creation.
 
 // CreateStoragePool is a function that does following:
-// 1. It receives storagepoolclaim object from the spc watcher event handler.
-// 2. After successful validation, it will call a worker function for actual storage creation
-//    via the cas template specified in storagepoolclaim.
-func (c *Controller) CreateStoragePool(spcGot *apis.StoragePoolClaim) error {
-	poolconfig := c.NewPoolCreateConfig(spcGot)
-	newCasPool, err := poolconfig.getCasPool(spcGot)
-
+// 1. It receives cstorpoolcluster object from the cspc watcher event handler.
+// 2. Call GetCasPool method from nodeselect package to get the CasPool object which is an internal representation
+//    of a cstor pool.
+func (po *PoolOperation) CreateStoragePool(cspcGot *apisv1alpha1.CStorPoolCluster) error {
+	newCasPool, err := po.GetCasPool()
```

Comment:
Does `NewCASPool()` is more close to what is happening vs. `GetCASPool()`
In addition, can caller write this as `po.nodeselect.NewCASPool()` This looks more explicit IMO.

Reply:
Good point. Will fix it.